.github/workflows/sync-registry.yaml

```diff
@@ -0,0 +1,18 @@
+on:
+  schedule:
+    - cron: "0 0 * * *" # Runs every midnight
+
+jobs:
+  publish:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Sync registry.k8s.io to ghcr.io/canonical
+        env:
+          ACTOR: ${{ github.actor }}
+          TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          DIR: ${GITHUB_WORKSPACE}
```

Comment:
no need to set DIR, see comment on bash script. the script can figure out its base directory on its own.